src/Entity/BannerSelection/Campaign.php

```diff
@@ -35,7 +35,22 @@ public function getEnd(): \DateTime {
 		return $this->end;
 	}
 
-	public function isActiveAtPointInTime( \DateTime $time ): bool {
-		return $time->getTimestamp() > $this->start->getTimestamp() && $time->getTimestamp() < $this->end->getTimestamp();
+	public function getStart(): \DateTime {
+		return $this->start;
+	}
+
+	public function isInActiveDateRange( \DateTime $time ): bool {
+		return
+			$time->getTimestamp() > $this->getStart()->getTimestamp() &&
```

Comment:
While it's common for a class to use its own getters (there are many examples in the OOP literature that do this), I think a class can get away with accessing its own properties directly. It avoids boilerplate code (99% of the getters return the property unchanged) and highlights the getters that really transform a property. Also, the existence of private getters carries the risk of someone saying "hey, i need this value, let's make the private into a public" instead of adding a meaningful interaction method to the class.

If using a property directly feels wrong to you (I know how wrong it felt to me the first time), you can leave it like this.